ui/widgets/tabs.js

```diff
@@ -270,7 +271,19 @@ $.widget( "ui.tabs", {
 			return index;
 		}
 
-		while ( $.inArray( constrain(), this.options.disabled ) !== -1 ) {
+		function canActivate( index ) {
+			if ( $.inArray( constrain(), that.options.disabled ) !== -1 ) {
+				return false;
+			}
+
+			if ( that.tabs.eq( index ).is( ":hidden" ) ) {
```

Comment:
When there are multiple failure cases, I generally like to list them out individually as failure cases, then return `true` at the end, but I'm fine making this change.